### PR TITLE
Annotations: Fix so adds, updates and deletes are shown correctly

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This PR fixes a bunch of different issues:
1. Previously if a user updated/added/deleted an annotation in the old graph panel, the result would not show until the user refreshed the dashboard
2. Previously if a user updated/added/deleted an annotation when the panel was in panel editor the annotation editor would user the id for the panel in the editor which is a random id assigned during panel edit

**Which issue(s) this PR fixes**:
Relates #35131

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by hugohaggmark** [comment]
@torkelo opened a smaller PR that handles just the add/update/delete of annotations in old graph. Will close the other until I can think of a better solution.

File: public/app/features/annotations/event_editor.ts [hunk 1]
```diff
@@ -48,39 +49,32 @@ export class EventEditorCtrl {
...
+    }
+
+    try {
+      await crudFunction(saveModel);
+    } catch (err) {
+      console.log(err);
+    } finally {
+      this.panelCtrl.refresh();
+      this.close();
+      getDashboardQueryRunner().run({ dashboard: this.panelCtrl.dashboard, range: this.panelCtrl.range });
```
Comment (torkelo) [inline public/app/features/annotations/event_editor.ts:64]: I think technically  getDashboardQueryRunner().run({ dashboard: this.panelCtrl.dashboard, range: this.panelCtrl.range });  should be enough here, issuing another metrics query should not be nessary?

Comment (torkelo) [reply public/app/features/annotations/event_editor.ts:64]: I mean if we remove `  this.panelCtrl.refresh();` this should still work

**Review by torkelo — APPROVED** [review]

File: public/app/features/annotations/event_editor.ts (hunk 1 above)
Comment (hugohaggmark) [reply public/app/features/annotations/event_editor.ts:64]: Ok, so we should remove `this.panelCtrl.refresh();`?

Comment (hugohaggmark) [reply public/app/features/annotations/event_editor.ts:64]: Never mind, saw your comment :)

**Review by torkelo — APPROVED** [review]

**Pull request merged**